golinks.go

```
@@ -1,156 +1,160 @@
-	flag.Parse()
-	log.Printf("starting golinks...")
-	r := NewRedirector(*configFile)
-	r.ReadConfig()
-
-	http.HandleFunc("/add/", r.AddLink)
-	http.HandleFunc("/list/", r.GetLinks)
-	http.HandleFunc("/del/", r.DelLink)
-	http.HandleFunc("/", r.Redirect)
-	if err := http.ListenAndServe(":"+*port, nil); err != nil {
-		log.Fatal("Unable to listen %s", err)
-	}
-	for {
-	}
+        port := flag.String("http_port", "8080", "Port number to listen")
+        configFile := flag.String("config", "redirects.json", "Configuration filename")
+        flag.Parse()
+        log.Printf("Starting golinks...")
+        r := newRedirector(*configFile)
+        r.readConfig()
+
+        http.HandleFunc("/add/", r.addLink)
+        http.HandleFunc("/list/", r.getLinks)
+        http.HandleFunc("/del/", r.delLink)
+        http.HandleFunc("/", r.redirect)
+        if err := http.ListenAndServe(":"+*port, nil); err != nil {
+                log.Fatal("Unable to listen %s", err)
+        }
 }
 
-func NewRedirector(file string) *Settings {
-	return &Settings{
-		Filename: file,
-	}
+func newRedirector(file string) *settings {
+        return &settings{
+                filename: file,
+        }
 }
 
-func (m *Settings) ReadConfig() {
-	log.Printf("reading configuration...")
-	jsonBlob, err := ioutil.ReadFile(m.Filename)
-	if err != nil {
-		log.Printf("Unable to read configuration %s", err)
-		return
-	}
-	if err := json.Unmarshal(jsonBlob, &m.Redirects); err != nil {
-		log.Printf("error unmarshalling %s", err)
-	}
+func (m *settings) readConfig() {
+        log.Printf("Reading configuration...")
+        jsonBlob, err := ioutil.ReadFile(m.filename)
+        if err != nil {
+                log.Printf("No config file found. Using new config file.")
+                return
+        }
+        if err := json.Unmarshal(jsonBlob, &m.redirects); err != nil {
+                log.Printf("Error unmarshalling %s", err)
+        }
 }
-
-func (m *Settings) SaveToDisk() {
```

this should probably return an error, then have the http handler provide an error message with http error code